app/contracts/work_packages/update_contract.rb

```diff
@@ -48,6 +48,10 @@ class UpdateContract < BaseContract
     default_attribute_permission :edit_work_packages
     attribute_permission :project_id, :move_work_packages
 
+    def can_set_parent?
+      @can.allowed?(model, :edit)
+    end
```

Comment:
I tested more and it looks like it's not enough: when I test with the edit_work_packages permission, it still fails, because one additional permission is needed: `:permission_manage_subtasks` ("Manage work package hierarchies")

And I also tried with only `:permission_manage_subtasks`, and it works! So it should check for `:manage_subtasks` permission instead.

And I tried also by sharing it with "Edit" level. It also does not display any error, but does not work. I think it will be fixed once `:manage_subtasks` permission is used.